Gordon360/Services/NewsService.cs

```diff
@@ -51,44 +51,63 @@ public StudentNews Get(int newsID)
 
         public async Task<IEnumerable<StudentNewsViewModel>> GetNewsNotExpiredAsync()
         {
-            var news = await _contextCCT.Procedures.NEWS_NOT_EXPIREDAsync();
-            return news.Select(n => new StudentNewsViewModel
-            {
-                SNID = n.SNID,
-                ADUN = n.ADUN,
-                categoryID = n.categoryID,
-                Subject = n.Subject,
-                Body = n.Body,
-                Image = n.Image,
-                Accepted = true,
-                Sent = n.Sent,
-                thisPastMailing = n.thisPastMailing,
-                Entered = n.Entered,
-                categoryName = n.categoryName,
-                SortOrder = n.SortOrder,
-                ManualExpirationDate = n.ManualExpirationDate,
-            });
+
+            var news = (from sn in _context.StudentNews
+                        join snc in _context.StudentNewsCategory
+                        on sn.categoryID equals snc.categoryID
+                        where sn.Accepted ?? false
+                        && ((sn.ManualExpirationDate == null
+                               && (sn.Entered ?? DateTime.Today).AddDays(14).Date >= (DateTime.Today).Date)
+                           || (sn.ManualExpirationDate != null
+                           && (sn.ManualExpirationDate ?? DateTime.Today).Date >= (DateTime.Today).Date))
```

Comment:
Great work on converting this logic to C#!

As we talked about in today's meeting, one other option for these date comparisons is to use SQL's Built-in [`DATEDIFF` function](https://learn.microsoft.com/en-us/sql/t-sql/functions/datediff-transact-sql?view=sql-server-ver16).

We can reference that function in C# using [`EF.Functions.DateDiffDay`](https://learn.microsoft.com/en-us/ef/core/providers/sql-server/functions#date-and-time-functions):

```suggestion
                        && ((sn.ManualExpirationDate == null
                               && EF.Functions.DateDiffDay(sn.Entered ?? DateTime.Today, DateTime.Today) < 14)
                           || (sn.ManualExpirationDate != null
                           && EF.Functions.DateDiffDay(sn.ManualExpirationDate ?? DateTime.Today, DateTime.Today) < 1))
```

As I mentioned, I'm not totally sold that this is better, but I wanted to demonstrate it as an option.